dfastmi/batch/core.py

```diff
@@ -571,102 +580,138 @@ def analyse_and_report(
-    apply_q : BoolVector
-        A tuple of 3 flags indicating whether each value should be used or not.
-    T : Vector
-        Fraction of year represented by each forcing condition.
-    rsigma : Vector
-        Array of relaxation factors; one per forcing condition.
-    slength : float
-        The expected yearly impacted sedimentation length.
-    ucrit : float
-        Critical flow velocity [m/s].
-    filenames : Dict[Any, Tuple[str,str]]
-        Dictionary of the names of the data file containing the simulation
-        results to be processed. The conditions (discharge, wave conditions,
-        ...) are the key in the dictionary. Per condition a tuple of two file
-        names is given: a reference file and a file with measure.
-    xykm : shapely.geometry.linestring.LineString
-        Original river chainage line.
-    needs_tide : bool
-        Specifies whether the tidal boundary is needed.
-    n_fields : int
-        Number of fields to process (e.g. to cover a tidal period).
-    tide_bc : Tuple[str, ...]
-        Array of tidal boundary condition; one per forcing condition.
-    old_zmin_zmax : bool
-        Specifies the minimum and maximum should follow old or new definition.
-    kmbounds : Tuple[float,float]
-        Minimum and maximum chainage values indicating range of interest.
+    rootdir : str
+        Reference directory for default folders.        
     outputdir : str
-        Name of the output directory.
-    plotops : Dict
-        Dictionary of plot settings
+        Reference directory for default output folders.
+    gui : bool
+        Flag indicating whether this routine is called from the GUI.
 
     Returns
     -------
     success : bool
         Flag indicating whether analysis could be carried out.
     """
+    initialized_config = ConfigurationInitializerFactory.generate(cfg_version, reach, config)
+    
+    old_zmin_zmax = False
+
+    # set plotting flags
+    plotops = _set_plotting_flags(rootdir, display, data)
+
+    imode = _log_report_mode_usage(config, report)
+    filenames = get_filenames(imode, initialized_config.needs_tide, config)
+    success = False
+
     if imode == 0:
-        return AnalyserAndReporterWaqua.analyse_and_report_waqua(
+        success = AnalyserAndReporterWaqua.analyse_and_report_waqua(
             display,
             report,
             reduced_output,
-            tstag,
-            Q,
-            apply_q,
-            T,
-            rsigma,
-            ucrit,
+            initialized_config.tstag,
+            initialized_config.discharges,
+            initialized_config.apply_q,
+            initialized_config.time_fractions_of_the_year,
+            initialized_config.rsigma,
+            initialized_config.ucrit,
             old_zmin_zmax,
             outputdir,
         )
     else:
-        return AnalyserAndReporterDflowfm.analyse_and_report_dflowfm(
+        success = AnalyserAndReporterDflowfm.analyse_and_report_dflowfm(
```

Comment:
some of the passed on parameters are unused

Reply:
I will tackle it during conflicts with 123

Reply:
Yes, that is always the case..